### PR TITLE
Intake error handling on Ramp Refiling Review page

### PR DESCRIPTION
**Pull request by mmlumba**
With both server error and ineligible for HLR error:
screen shot 2018-01-17 at 2 46 18 pm

Removing HLR error on selection of another option:
intake-error-hlr

Connects #4216

### CODE REVIEW

**Comment by mmlumba** [comment]
@abbyraskinUSDS Please take a look at the first screenshot and let me know if that looks good

File: client/app/intake/pages/rampRefiling/review.jsx [hunk 1]
```diff
@@ -59,19 +60,25 @@ class Review extends React.PureComponent {
     ];
 
     return <div>
-      { hasInvalidOption && <Alert title="Ineligible for Higher-Level Review" type="error" lowerMargin>
+      { hasInvalidOption &&
+        Boolean(optionSelected.match(/(higher_level_review)\w*/g)) &&
```
Comment (shanear) [inline client/app/intake/pages/rampRefiling/review.jsx]: This is a tad complicated, and could break if we ever change the keys for the options. It'd be more straightforward, error proof, and readable if we just matched to the constants

```
optionSelected === REVIEW_OPTIONS.HIGHER_LEVEL_REVIEW.key ||
optionSelected === REVIEW_OPTIONS.HIGHER_LEVEL_REVIEW_WITH_HEARING.key
```

Additionally, we should move this logic to the reducer, and set `hasInvalidOption` when optionSelected is changing based on what value it is.

File: client/app/intake/pages/rampRefiling/review.jsx [hunk 2]
```diff
@@ -120,7 +127,8 @@ export default connect(
     requestStatus: state.rampRefiling.requestStatus,
     intakeId: state.intake.id,
     appealDocket: state.rampRefiling.appealDocket,
-    appealDocketError: state.rampRefiling.appealDocketError
+    appealDocketError: state.rampRefiling.appealDocketError,
+    beginNextIntakeError: state.rampRefiling.beginNextIntakeError
```
Comment (shanear) [inline client/app/intake/pages/rampRefiling/review.jsx]: This name is sort of confusing in the global context of things. I would think that it's referring to the begin next intake button on the complete screen. Maybe rename to `submitInvalidOptionError`?

File: client/app/intake/pages/rampRefiling/review.jsx [hunk 3]
```diff
@@ -146,7 +154,8 @@ class ReviewNextButton extends React.PureComponent {
       onClick={this.handleClick}
       loading={this.props.requestState === REQUEST_STATE.IN_PROGRESS}
       legacyStyling={false}
-      disabled={this.props.hasInvalidOption}
+      disabled={Boolean(this.props.hasInvalidOption &&
+        this.props.optionSelected.match(/(higher_level_review)\w*/g))}
```
Comment (shanear) [inline client/app/intake/pages/rampRefiling/review.jsx]: Again, if we move the logic to the reducer this won't be necessary.

**Pull request merged**